VPC-Section-Formatting (#141)
C2Devel/terraform-provider-croccloud: docs/data-sources/route_table.md

Reviewer: м.б. эту формулировку адаптировать к предыщущей route.md

Reviewer: available?

Reviewer: Здесь выглядит как масло масляное (ниже у каждого аргумента указывается, что он Optional)
возможны 2 решения
1) если мы все равно разбиваем аргументы на обязательные и оппциональные, то какой смысл указывать это еще доп. в скобках?
2) интересное решение, кажется у Яндекса -- они все аргументы дают по алфавиту, а слово Required выделяют жирным, чтобы сразу в глаза бросалось
@PixarV  что думаешь?

Author: @DiGa66 @PixarV 
у Яндекса оно выглядит некрасиво) разбиение удобно тем, что если мы добавим новый required или optional аргумент, то сможем добавить его в нужную секцию, и будет выглядеть норм

Reviewer 2: Фразы `The following arguments are required/optional` в ресурсах с обновленной документацией не используются. В основном используется общий список: сначала required, потом optional. 

Если переходить на использование `The following arguments are required/optional` по всей доке, то придется добавлять эти строчки в том числе на уровень блоков. Возможно будет смотреться громоздко.

Мне тоже кажется, что не имеет смысла указывать опциональность, если уже сказано, что все аргументы в списке опциональные.

Reviewer 2: Выделение жирным не очень нравится.

Provides information about a route table.

> It can be used when a module accepts a subnet ID as an input variable and needs to, for example, add a route to a route table.

Reviewer: ```suggestion
The data source can be used when a module accepts a subnet ID as an input variable and needs to, for example, add a route to a route table.
```

Provides information about a route table.

> This resource can be used when a module accepts a subnet ID as an input variable and needs to, for example, add a route to the route table.

Reviewer: см коменты по поводу route, 
и здесь a route table (таблица тоже не определена)

}
```

## Argument reference

The arguments of this data source act as filters for querying route tables in the current region.

~> **Note** The given filters must exactly match the resource whose data will be exported as attributes.

The following arguments are optional:

* `filter` - (Optional, [Block](#filter)) One or more name/value pairs to use as filters.
    * _Valid values:_ See supported names and values in [EC2 API documentation][describe-route-tables]
* `subnet_id` - (Optional, String) The ID of the subnet which is associated with the route table (not exported if not passed as a parameter).
* `tags` - (Optional, Map of strings) Key-value pairs. Must exactly match pairs on the required resource.
* `vpc_id` - (Optional, String) The ID of the VPC that owns the desired route table.

### filter

* `name` - (Required, String) The name of the filter.
    * _Constraints:_ Filter names are case-sensitive
* `values` - (Required, List of strings) One or more filter values.
    * _Constraints:_ Filter values are case-sensitive

## Attribute reference

> ### Supported attributes

Reviewer 2: Докинь в атрибуты еще пожалуйста `route_table_id` и `id` (оба содержат id таблицы)

Author: в какой-то блок или в основной список? `route_table_id` есть в блоке `associations`
@PixarV

Reviewer 2: В основной список. `route_table_id` есть и в основном списке, и в блоке `associations`

In addition to all arguments above, the following attributes are exported:

* `arn` - (String) The Amazon Resource Name (ARN) of the route table.
* `associations` - ([Block](#associations)) The list of associations.
* `routes` - ([Block](#routes)) The list of routes.

#### associations

> * `gateway_id` - (String) The ID of the internet gateway.

Reviewer 2: Перемести пожалуйста в неподдерживаемые

* `main` - (Boolean) Indicates whether the route table is the main route table for the VPC.
* `route_table_association_id` - (String) The ID of the route table association.
* `route_table_id` - (String) The ID of the route table.
* `subnet_id` - (String) The ID of the subnet the route table is associated with. Only set when associated with a subnet.

#### routes

* `cidr_block` - (String) The CIDR block of the route.
* `gateway_id` - (String) The ID of the internet gateway.
* `instance_id` - (String) The ID of the instance.
* `network_interface_id` - (String) The ID of the network interface.
* `transit_gateway_id` - (String) The ID of the transit gateway.

> ### Unsupported attributes

Reviewer 2: Добавь пожалуйста в неподдерживаемые еще `gateway_id`

}
```

## Argument reference

The arguments of this data source act as filters for querying route tables in the current region.

> The given filters must exactly match one route table whose data will be exported as attributes.

Reviewer 2: Стандартная формлуировка + инфо/варнинг

The following arguments are optional:

* `filter` - (Optional, [Block](#filter)) One or more name/value pairs to use as filters.
    * _Valid values:_ See supported names and values in [EC2 API documentation][describe-route-tables]
* `subnet_id` - (Optional, String) The ID of the subnet which is associated with the route table (not exported if not passed as a parameter).
* `tags` - (Optional, Map of strings) Key-value pairs. Must exactly match pairs on the required resource.
* `vpc_id` - (Optional, String) The ID of the VPC that the desired route table belongs to.

### filter

* `name` - (Required, String) The name of the filter. Names are case-sensitive.
* `values` - (Required, List of strings) One or more filter values. The values are case-sensitive.

## Attribute reference

### Supported attributes

In addition to all arguments above, the following attributes are exported:

* `arn` - (String) The Amazon Resource Name (ARN) of the route table.
* `associations` - ([Block](#associations)) The list of associations.
* `routes` - ([Block](#routes)) The list of routes.

#### associations

> Associations are also exported with the following attributes:

Reviewer 2: На страничках выше такого замечания для блоков нет. Нужным не выглядит

* `gateway_id` - (String) The ID of the internet gateway.
* `main` - (Boolean) Indicates whether the route table is the main route table for the VPC.
* `route_table_association_id` - (String) The ID of the route table association.
* `route_table_id` - (String) The ID of the route table.
* `subnet_id` - (String) The ID of the subnet the route table is associated with. Only set when associated with a subnet.

#### routes

When relevant, routes are also exported with the following attributes:

> For destinations:

Reviewer 2: Можно убрать и сделать обычным списком атрибутов

* `arn` - (String) The Amazon Resource Name (ARN) of the route table.
* `id` - (String) The ID of the route table.
* `associations` - ([Block](#associations)) The list of associations.
* `route_table_id` - (String) The ID of the route table.
* `routes` - ([Block](#routes)) The list of routes.

#### associations

* `main` - (Boolean) Indicates whether the route table is the main route table for the VPC.
* `route_table_association_id` - (String) The ID of the route table association.
* `route_table_id` - (String) The ID of the route table.
* `subnet_id` - (String) The ID of the subnet the route table is associated with. Only set when associated with a subnet.

#### routes

* `cidr_block` - (String) The CIDR block of the route.
* `instance_id` - (String) The ID of the instance.
* `network_interface_id` - (String) The ID of the network interface.
* `transit_gateway_id` - (String) The ID of the transit gateway.

### Unsupported attributes

~> **Note** These attributes may be present in the `terraform.tfstate` file, but they have preset values and cannot be specified in configuration files.

The following attributes are not currently supported:

> `associations.gateway_id`, `owner_id`, `routes.carrier_gateway_id`, `routes.core_network_arn`, `routes.destination_prefix_list_id`, `routes.egress_only_gateway_id`, `routes.gateway_id`, `routes.ipv6_cidr_block`, `routes.local_gateway_id`, `routes.nat_gateway_id`, `routes.vpc_endpoint_id`, `routes.vpc_peering_connection_id`.

Reviewer 2: ```suggestion
`associations.gateway_id`, `gateway_id`, `owner_id`, `routes.carrier_gateway_id`, `routes.core_network_arn`, `routes.destination_prefix_list_id`, `routes.egress_only_gateway_id`, `routes.ipv6_cidr_block`, `routes.local_gateway_id`, `routes.nat_gateway_id`, `routes.vpc_endpoint_id`, `routes.vpc_peering_connection_id`.
```

`routes.gateway_id` - поддерживаемый

## Argument Reference

The arguments of this data source act as filters for querying route tables in the current region. The given filters must match exactly one route table whose data will be exported as attributes.

The following arguments are optional:

* `filter` - (Optional, List of strings) One or more name/value pairs to use as filters.
    * _Valid values:_ See supported names and values in [EC2 API documentation][describe-route-tables]
* `subnet_id` - (Optional, String) The ID of the subnet which is associated with the route table (not exported if not passed as a parameter).
* `tags` - (Optional, Map of strings) Key-value pairs. Must exactly match pairs on the required resource.
* `vpc_id` - (Optional, String) The ID of the VPC that the desired route table belongs to.

## Attribute Reference

### Supported attributes

In addition to all arguments above, the following attributes are exported:

* `arn` - (String) The Amazon Resource Name (ARN) of the route table.

> * `associations` - The list of associations.

Reviewer: Тип?

Author: Тип указан, коммент почему-то не зарезолвлен

  See [associations](#associations) for more details.

> * `routes` - The list of routes.

Reviewer: Тип?

  See [routes](#routes) for more details.

#### associations

Associations are also exported with the following attributes:

* `gateway_id` - (String) The ID of the internet gateway.
* `main` - (Boolean) Indicates whether the route table is the main route table for the VPC.
* `route_table_association_id` - (String) The ID of the route table association.
* `route_table_id` - (String) The ID of the route table.
* `subnet_id` - (String) The ID of the subnet the route table is associated with. Only set when associated with a subnet.

#### routes

When relevant, routes are also exported with the following attributes:

For destinations:

* `cidr_block` - (String) The CIDR block of the route.

For targets:

> * `gateway_id` - (String) The ID of the internet gateway..

Reviewer: ```suggestion
* `gateway_id` - (String) The ID of the internet gateway.
```

* `filter` - (Optional) One or more name/value pairs to use as filters.
    * _Valid values:_ See supported names and values in [EC2 API documentation][describe-route-tables]
* `subnet_id` - (Optional) (String) The ID of a subnet which is associated with the route table (not exported if not passed as a parameter).
* `tags` - (Optional) (Map of tags) Key-value map of tags, each pair of which must exactly match a pair on the desired route table.

> * `vpc_id` - (Optional) (string) The ID of the VPC that the desired route table belongs to.

Reviewer: ```suggestion
* `vpc_id` - (Optional, String) The ID of the VPC that the desired route table belongs to.
```

## Attribute Reference

### Supported attributes

In addition to all arguments above, the following attributes are exported:

* `arn` - (String) The Amazon Resource Name (ARN) of the route table.
* `associations` - The list of associations.
  The structure of this block is [described below](#associations).
* `routes` - The list of routes.
  The structure of this block is [described below](#routes).

#### associations

Associations are also exported with the following attributes:

* `gateway_id` - (String) The ID of the internet gateway.

> * `main` - (Bool) Indicates whether the route table is the main route table for the VPC.

Reviewer: Boolean (полностью, как в пользовательской доке)

Author: в оригинале Bool
https://developer.hashicorp.com/terraform/plugin/framework/handling-data/types#primitive-types
@DiGa66

> * `route_table_association_id` - (String) The route table association ID.

Reviewer: ```suggestion
* `route_table_association_id` - (String) The ID of the route table association.
```

* `route_table_id` - (String) The ID of the route table.
* `subnet_id` - (String) The ID of the subnet the route table is associated with. Only set when associated with a subnet.

#### routes

When relevant, routes are also exported with the following attributes:

For destinations:

* `cidr_block` - (String) The CIDR block of the route.

For targets:

> * `gateway_id` - (String) The ID of the internet gateway or virtual private gateway.

Reviewer: virtual private gateway у нас такого вроде нет
